src/geometry/Point.js

```diff
@@ -25,10 +25,13 @@ import {isArray, formatNum} from '../core/Util';
  */
 
 export function Point(x, y, round) {
+	if (isNaN(x) || isNaN(y)) {
+		throw new Error('Invalid Point object: (' + x + ', ' + y + ')');
+	}
 	// @property x: Number; The `x` coordinate of the point
-	this.x = (round ? Math.round(x) : x);
+	this.x = (round ? Math.round(x) : parseFloat(x));
 	// @property y: Number; The `y` coordinate of the point
-	this.y = (round ? Math.round(y) : y);
+	this.y = (round ? Math.round(y) : parseFloat(y));

```

Comment:
I also worry about values like `'0x10'` that will be converted to `0`.
And it's not clear what should we do with "Infinity".

See this table:
![image](https://i.stack.imgur.com/QNMfI.png)
(from https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which)